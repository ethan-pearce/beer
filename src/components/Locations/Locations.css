.location{
    margin: 16px;
    border: 1px solid;
}

.location:hover{
    margin: 16px;
    border: 1px solid lightgray;
}

.locations{
    display: grid;
    grid-template-columns: repeat(4, 25%);
}

@media only screen and (max-width: 1023px) {
    .locations {
        display: grid;
        grid-template-columns: repeat(2, 50%);
    }
  }

.LocationTitle {
    text-align: left;
    padding: 0 16px;
    font-weight: 200;
    font-size: 24px;
    margin: 16px 0 0;
}