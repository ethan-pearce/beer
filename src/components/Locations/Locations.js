import './Locations.css';

export const Locations = ({locations, breweryName}) => {
    if(!locations || locations.length <= 0){
      return null;
    }

    return (
      <>
        <p className={'LocationTitle'}>Locations:</p>
        <ul className={'locations'}>
          {locations.map(location => {
            const { id, name, type, region, postalCode } = location;
            return (
              <li key={id} className={'location'}>
                <a href={`https://www.google.com/maps/search/?api=1&query=${breweryName}+${name}+${region}+${postalCode}`} rel="noreferrer" target="_blank">
                  <p>{name}</p>
                  <p>{`Type: ${type}`}</p>
                  <p>{region}</p>
                  <p>{postalCode}</p>
                </a>
              </li>
            )
          })}
        </ul>
      </>
    )
  }

  export default Locations;