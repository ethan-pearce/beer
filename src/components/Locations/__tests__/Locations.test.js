import React from 'react'
import { render } from '@testing-library/react';
import Locations from '../Locations'

describe('Location Component', () => {
    const locations = [
        {
            name: 'test name 1',
            id: '1',
            type: 'test type 1',
            region: 'test region 1',
            postalCode: '00001'
        },
        {
            name: 'test name 2',
            id: '2',
            type: 'test type 2',
            region: 'test region 2',
            postalCode: '00002'
        },
    ];

    it('should render', () => {
        const { container } = render(<Locations locations={locations}/>)

        expect(container.getElementsByClassName('LocationTitle').length).toBe(1)
        expect(container.getElementsByClassName('location').length).toBe(2)
        expect(container).toMatchSnapshot()
    })

    it('should return null if not locations passed', () => {
        const { container } = render(<Locations />)

        expect(container.getElementsByClassName('LocationTitle').length).toBe(0)
        expect(container.getElementsByClassName('location').length).toBe(0)
        expect(container).toMatchSnapshot()
    })
})