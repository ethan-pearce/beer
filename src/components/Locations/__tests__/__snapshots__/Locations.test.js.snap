// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Component should render 1`] = `
<div>
  <p
    class="LocationTitle"
  >
    Locations:
  </p>
  <ul
    class="locations"
  >
    <li
      class="location"
    >
      <a
        href="https://www.google.com/maps/search/?api=1&query=undefined+test name 1+test region 1+00001"
        rel="noreferrer"
        target="_blank"
      >
        <p>
          test name 1
        </p>
        <p>
          Type: test type 1
        </p>
        <p>
          test region 1
        </p>
        <p>
          00001
        </p>
      </a>
    </li>
    <li
      class="location"
    >
      <a
        href="https://www.google.com/maps/search/?api=1&query=undefined+test name 2+test region 2+00002"
        rel="noreferrer"
        target="_blank"
      >
        <p>
          test name 2
        </p>
        <p>
          Type: test type 2
        </p>
        <p>
          test region 2
        </p>
        <p>
          00002
        </p>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Location Component should return null if not locations passed 1`] = `<div />`;
